test/e2e/network_segmentation.go

```diff
@@ -1067,16 +1067,19 @@ spec:
 			return nil
 		})
 
-		conditionsJSON, err := e2ekubectl.RunKubectl(f.Namespace.Name, "get", "clusteruserdefinednetwork", cudnName, "-o", "jsonpath={.status.conditions}")
-		Expect(err).NotTo(HaveOccurred())
-		var actualConditions []metav1.Condition
-		Expect(json.Unmarshal([]byte(conditionsJSON), &actualConditions)).To(Succeed())
-
-		Expect(actualConditions[0].Type).To(Equal("NetworkReady"))
-		Expect(actualConditions[0].Status).To(Equal(metav1.ConditionFalse))
-		Expect(actualConditions[0].Reason).To(Equal("NetworkAttachmentDefinitionSyncError"))
 		expectedMessage := fmt.Sprintf("primary network already exist in namespace %q: %q", primaryNetTenantNs, primaryNadName)
-		Expect(actualConditions[0].Message).To(Equal(expectedMessage))
+		Eventually(func() []metav1.Condition {
+			conditionsJSON, err := e2ekubectl.RunKubectl(f.Namespace.Name, "get", "clusteruserdefinednetwork", cudnName, "-o", "jsonpath={.status.conditions}")
+			Expect(err).NotTo(HaveOccurred())
+			var actualConditions []metav1.Condition
+			Expect(json.Unmarshal([]byte(conditionsJSON), &actualConditions)).To(Succeed())
+			return normalizeConditions(actualConditions)
+		}, 5*time.Second, 1*time.Second).Should(Equal([]metav1.Condition{{
+			Type:    "NetworkReady",
+			Status:  metav1.ConditionFalse,
+			Reason:  "NetworkAttachmentDefinitionSyncError",
+			Message: expectedMessage,
+		}}))
```

Comment:
if you're going to use expectations within eventually / consistently blocks, please use `GinkgoHelper` and pass gomega instance as the first function attribute.

Copying from ginkgo's documentation:
> Eventually can also accept functions that take a Gomega parameter. These functions are then passed a local Gomega that can be used to make assertions inside the function as it is polled. Eventually will retry the function if an assertion fails. This would allow us to replace:

```suggestion
		Eventually(func(g Gomega) []metav1.Condition {
			GinkgoHelper()
			conditionsJSON, err := e2ekubectl.RunKubectl(f.Namespace.Name, "get", "clusteruserdefinednetwork", cudnName, "-o", "jsonpath={.status.conditions}")
			g.Expect(err).NotTo(HaveOccurred())
			var actualConditions []metav1.Condition
			g.Expect(json.Unmarshal([]byte(conditionsJSON), &actualConditions)).To(Succeed())
			return normalizeConditions(actualConditions)
		}, 5*time.Second, 1*time.Second).Should(Equal([]metav1.Condition{{
			Type:    "NetworkReady",
			Status:  metav1.ConditionFalse,
			Reason:  "NetworkAttachmentDefinitionSyncError",
			Message: expectedMessage,
		}}))
```

Reply:
For fetching the object (line 1072) it make sense because there could be some slow down with KAPI.
But if unmarshaling (line 1075) fails we have nothing to do, and I dont mind if the test will blow up. WDYT?

Reply:
How can unmarshalling something that comes from the k8s API fail ? You're asserting that for completeness AFAIU.

And still, the argument still applies: it a next reconcile cycle, it could have been "fixed".

Reply:
Done